admin-auth/src/main/kotlin/com/epam/drill/admin/auth/service/impl/JwkTokenService.kt

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2020 - 2022 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.epam.drill.admin.auth.service.impl
+
+import com.auth0.jwk.Jwk
+import com.auth0.jwk.JwkException
+import com.auth0.jwk.JwkProvider
+import com.auth0.jwk.JwkProviderBuilder
+import com.auth0.jwt.JWT
+import com.auth0.jwt.algorithms.Algorithm
+import com.auth0.jwt.exceptions.JWTDecodeException
+import com.auth0.jwt.exceptions.JWTVerificationException
+import com.auth0.jwt.interfaces.JWTVerifier
+import com.epam.drill.admin.auth.config.OAuthConfig
+import com.epam.drill.admin.auth.model.UserView
+import com.epam.drill.admin.auth.service.TokenService
+import io.ktor.auth.jwt.*
+import java.net.URL
+import java.security.interfaces.ECPublicKey
+import java.security.interfaces.RSAPublicKey
+import java.util.concurrent.TimeUnit
+
+class JwkTokenService(oauthConfig: OAuthConfig) : TokenService {
```

Comment:
`JwkTokenService` -> `OauthJwkTokenService`?

Reply:
This code will be refactored. The entire class will likely be removed.